### PR TITLE
Fix bug in actor_counters.swift test

### PR DESCRIPTION
**Pull request by yln**
`UnsafeMutableBufferPointer.allocate` returns uninitialized memory which
is read by `assert(scratchBuffer[current] == 0)`.  We need to initialize
it before reading.

Also remove the scratchBuffer that is never used and likely a
refactoring leftover.

Radar-Id: rdar://72439642

### CODE REVIEW

File: test/Concurrency/Runtime/actor_counters.swift [hunk 1]
```diff
@@ -4,9 +4,6 @@
 // REQUIRES: concurrency
 // REQUIRES: libdispatch
 
-// Remove with rdar://problem/72439642
-// UNSUPPORTED: asan
-
```
Comment (yln) [inline test/Concurrency/Runtime/actor_counters.swift:9]: This test was disabled under ASan, but really only worked "by accident" in non-ASan configurations.

**Comment by yln** [comment]
@swift-ci please smoke test

**Comment by nate-chandler** [comment]
@swift-ci Please ASAN test

**Review by DougGregor — APPROVED** [review]
Thank you!

**Review by ktoso — APPROVED** [review]
LGTM, thanks!

The CI failures on linux and windows are spurious, I'll restart and this should be able to get in 👍

**Comment by ktoso** [comment]
@swift-ci please smoke test linux

**Pull request merged**